services/QuillLMS/app/workers/sync_vitally_past_year_users_worker.rb

```diff
@@ -0,0 +1,10 @@
+class SyncVitallyPastYearUsersWorker
+  include Sidekiq::Worker
+
+  def perform(teacher_ids, year)
+    teachers = User.where(id: teacher_ids)
+    teachers.each do |teacher|
```

Comment:
I would create a new worker here:
`CachePreviousYearUserDatumWorker.perform_async`. This keeps the individual jobs very small but also if one of them fails it doesn't restart the whole SyncVitallyPastYearUsersWorker